openwisp_radius/utils.py

```diff
@@ -130,27 +130,35 @@ def find_available_username(username, users_list, prefix=False):
     return tmp
 
 
-def decode_csv_data(csv_data):
-    if isinstance(csv_data, bytes):
-        # Detect encoding using chardet
-        detected = chardet.detect(csv_data)
-        detected_encoding = detected.get('encoding')
-
-        # Explicit handling for UTF-16 encodings (check for BOM)
-        if csv_data.startswith(b'\xff\xfe') or csv_data.startswith(b'\xfe\xff'):
-            detected_encoding = 'utf-16'
-        detected_encoding = detected_encoding or 'utf-8'
+
+def get_encoding_format(csv_data):
+    
+    # Explicit handling for UTF-16 encodings (check for BOM)
+    if csv_data.startswith(b'\xff\xfe') or csv_data.startswith(b'\xfe\xff'):
+        return 'utf-16'
+
+    # Detect encoding using chardet
+    detected = chardet.detect(csv_data)
+    detected_encoding = detected.get('encoding')
+
+    if detected_encoding == 'ascii': 
+        return 'utf-8'
+    
+    if detected_encoding == 'utf-16le':
         try:
-            return csv_data.decode(detected_encoding)
-        except UnicodeDecodeError as e:
-            raise
-        return csv_data
+            csv_data.decode('utf-16le')  # Test if decoding works
+            return detected_encoding
+        except UnicodeDecodeError:
+            pass
+        
+    return detected_encoding or 'utf-8'
 
 
 def validate_csvfile(csvfile):
     csv_data = csvfile.read()
     try:
-        csv_data = decode_csv_data(csv_data)
+        if isinstance(csv_data, bytes): 
+            csv_data = csv_data.decode(get_encoding_format(csv_data))  
```

Comment:
what happens if csv_data it's not an instance of bytes?

Reply:
then, decoding the data is not needed. so we do nothing.
actually previously 
```csv_data = csv_data.decode('utf-8') if isinstance(csv_data, bytes) else csv_data```
logic is still same.